utils/minnesota.py

```diff
@@ -0,0 +1,414 @@
+import uuid
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+from clean import StateCleaner
+from constants import (
+    MN_CANDIDATE_CONTRIBUTION_COL,
+    MN_CANDIDATE_CONTRIBUTION_MAP,
+    MN_INDEPENDENT_EXPENDITURE_COL,
+    MN_INDEPENDENT_EXPENDITURE_MAP,
+    MN_NONCANDIDATE_CONTRIBUTION_COL,
+    MN_NONCANDIDATE_CONTRIBUTION_MAP,
+    MN_RACE_MAP,
+)
+
+here = Path(__file__).resolve()
+repo_root = here.parent.parent
+
+FILEPATHS_LST = [
+    repo_root / "data" / "AG.csv",
+    repo_root / "data" / "AP.csv",
+    repo_root / "data" / "DC.csv",
+    repo_root / "data" / "GC.csv",
+    repo_root / "data" / "House.csv",
+    repo_root / "data" / "SA.csv",
+    repo_root / "data" / "SC.csv",
+    repo_root / "data" / "Senate.csv",
+    repo_root / "data" / "SS.csv",
+    repo_root / "data" / "ST.csv",
+    repo_root / "data" / "non_candidate_con.csv",
+    repo_root / "data" / "independent_exp.csv",
+]
+
+
+class MinnesotaCleaner(StateCleaner):
+    def entity_name_dictionary(self) -> dict:
+        self.entity_name_dictionary = {
+            "I": "Individual",
+            "L": "Lobbyist",
+            "B": "Company",
+            "PCF": "Committee",
+            "C": "Committee",
+            "F": "Committee",
+            "H": "Committee",
+            "S": "Committee",
+            "U": "Committee",
+            "PTU": "Party",
+            "P": "Party",
+            "O": "Other",
+        }
+
+        return self.entity_name_dictionary
+
+    def preprocess_candidate_contribution(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Helper function to preprocess MN candidate-recipient contribution df
+
+        Args:
+            df (DataFrame): a single raw MN candidate contribution df
+        Returns:
+            DataFrame: Preprocessed contribution df of candidate recipients
+        """
+
+        df1 = df.copy(deep=True)
+        df1 = df1[MN_CANDIDATE_CONTRIBUTION_COL]
+        df1 = df1.rename(columns=MN_CANDIDATE_CONTRIBUTION_MAP)
+        df1["recipient_type"] = "I"
+        df1["recipient_full_name"] = np.nan
```

Comment:
convert to None, since np.nan doesn't represent missing data, and it is difficult to work with